plugins/package-managers/cocoapods/src/main/kotlin/PodDependencyHandler.kt

```diff
@@ -111,6 +116,48 @@ internal class PodDependencyHandler : DependencyHandler<Lockfile.Pod> {
         }
     }
 
+    private fun File.parsePodspecFile(): Podspec? {
+        val content = readText()
+
+        return if ("Pod::Spec.new" in content) {
+            convertRubyPodspecFile(content)
+        } else {
+            content
+        }?.parsePodspec()
+    }
+
+    private fun File.convertRubyPodspecFile(content: String): String? {
+        // The podspec is in Ruby format.
+        // Because it may depend on React Native functions, an extra require may have to be injected.
+        val rubyContent = parentFile.searchUpwardsForSubdirectory("node_modules")?.let { nodeModulesParentDir ->
+            val reactNativePath =
+                nodeModulesParentDir.resolve("node_modules/react-native/scripts/react_native_pods.rb")
+            if (reactNativePath.isFile) {
+                "require '$reactNativePath'\n$content"
+            } else {
+                null
+            }
+        } ?: content
+
+        val tempRubyPath = createTempFile("ruby_podspec", ".podspec").apply { writeText(rubyContent) }
+
+        // Convert the Ruby podspec file to JSON.
+        val podspecProcess = CocoaPodsCommand.run(
+            parentFile, "ipc", "spec", tempRubyPath.pathString
+        )
+
+        return if (podspecProcess.isError) {
+            logger.warn {
+                "Failed to process the '.podspec' file in Ruby format at '$canonicalPath': " +
+                    podspecProcess.errorMessage
+            }
+
+            null
+        } else {
+            podspecProcess.stdout
+        }
+    }
```

Comment:
Maybe this can be a bit more readable?
```
return runCatching {
   // Convert the Ruby podspec file to JSON.
   val process = CocoaPodsCommand.run(parentFile, "ipc", "spec", tempRubyPath.pathString)
          .requireSuccess()
          .stdout
}.onFailure { e ->
    logger.warn {
        "Failed to process the '.podspec' file in Ruby format at '$canonicalPath': ${e.message.orEmpty()}"
            podspecProcess.errorMessage
    }
}.getOrNull()
```

Reply:
You cannot access `process.errorMessage` in `onFailure`, as `process` is scoped to `runCatching`.

Reply:
> You cannot access `process.errorMessage` in `onFailure`, as `process` is scoped to `runCatching`.

This was a mistake, updated above code.

Reply:
However, if using that code, `process` should probably be renamed to `spec` or `json`, as it does not contain the process object, but the JSON string.

Reply:
I remove the `process` variable as it was unused.